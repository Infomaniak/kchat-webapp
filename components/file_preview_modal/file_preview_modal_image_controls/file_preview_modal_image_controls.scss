.image-controls {
    display: flex;
    position: relative;
    color: white;
    height: 48px;
    background: rgba(61, 60, 64, 1);
    align-items: center;
    justify-content: center;
}

.image-controls-dropdown {
    height: 32px;
    font-weight: 400;
    font-size: 12px;
    line-height: 16px;
    width: 140px;
    color: white;
    border: none;
    border-bottom: 1px solid white;
    border-bottom-left-radius: 4px 1px;
    border-bottom-right-radius: 4px 1px;
    background-color: transparent;
}

#rectangle {
    width: 140px;
    height: 50px;
    border-radius: 5%;
    background-color: white;
}

.image-controls-button {
    font-family: 'compass-icons';
    font-style: normal;
    font-weight: 400;
    font-size: 18px;
    line-height: 18px;

    background-color: transparent;
    background-repeat: no-repeat;
    border: none;
    overflow: hidden;
    outline: none;
    padding: 0 22px 0 0;
    text-align: center;

    background-color: transparent;

    align-items: center;
    text-align: center;

    &.active {
        color: white;
    }

    &.inactive {
        color: rgba(255, 255, 255, 0.56);
    }
}
