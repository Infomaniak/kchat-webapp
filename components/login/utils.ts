// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
/* eslint-disable no-console */

import crypto from 'crypto';

import {Client4} from 'mattermost-redux/client';
import {IKConstants} from 'utils/constants-ik';
import {redirectUserToDefaultTeam} from 'actions/global_actions';
import {isServerVersionGreaterThanOrEqualTo} from 'utils/server_version';
import {getDesktopVersion} from 'utils/user_agent';

// eslint-disable-next-line no-process-env
const v2DefaultAuthServer = process.env.BASE_URL;

/**
 * Store IKToken infos in localStorage and update Client
 */
export function storeTokenResponse(response: { expires_in?: number; access_token: string; refresh_token?: string }) {
    // TODO: store in redux
    const d = new Date();
    d.setHours(d.getHours() + 2);

    localStorage.setItem('IKToken', response.access_token);

    if (response.refresh_token) {
        localStorage.setItem('IKRefreshToken', response.refresh_token);
    }

    if (response.expires_in) {
        localStorage.setItem('IKTokenExpire', parseInt(d.getTime() / 1000, 10));
        localStorage.setItem('tokenExpired', '0');
    }

    Client4.setToken(response.access_token);
    Client4.setCSRF(response.access_token);
    Client4.setAuthHeader = true;

    console.log('[login/utils > storeTokenResponse] new token stored at: ', d);
    Client4.setWebappVersion(GIT_RELEASE);
}

/**
 * Clear IKToken informations in localStorage
 */
export function clearLocalStorageToken() {
    localStorage.removeItem('IKToken');
    localStorage.removeItem('IKRefreshToken');
    localStorage.removeItem('IKTokenExpire');
    localStorage.setItem('tokenExpired', '1');
    window.postMessage(
        {
            type: 'token-cleared',
            message: {
                token: null,
            },
        },
        window.origin,
    );
    console.log('[login/utils > clearLocalStorageToken] token storage cleared at: ', new Date());
}

/**
 * get code_verifier for challenge
 * @returns string
 */
export function getCodeVerifier() {
    const ramdonByte = crypto.randomBytes(33);
    const hash =
               crypto.createHash('sha256').update(ramdonByte).digest();
    return hash.toString('base64').
        replace(/\+/g, '-').
        replace(/\//g, '_').
        replace(/[=]/g, '');
}

/**
 * Generate code_challenge for oauth
 * @param codeVerifier string
 * @returns string
 */
export async function generateCodeChallenge(codeVerifier: string) {
    const hash =
           crypto.createHash('sha256').update(codeVerifier).digest();
    return hash.toString('base64').
        replace(/\+/g, '-').
        replace(/\//g, '_').
        replace(/[=]/g, '');
}

/**
 * get code_challenge and redirect to IK Login
 */
export function getChallengeAndRedirectToLogin(infinite = false) {
    const redirectTo = window.location.origin.endsWith('/') ? window.location.origin : `${window.location.origin}/`;
    const codeVerifier = getCodeVerifier();
    let codeChallenge = '';

    generateCodeChallenge(codeVerifier).then((challenge) => {
        codeChallenge = challenge;

        // TODO: store in redux instead of localstorage
        localStorage.setItem('challenge', JSON.stringify({verifier: codeVerifier, challenge: codeChallenge}));

        window.location.assign(`${IKConstants.LOGIN_URL}authorize?code_challenge=${codeChallenge}${infinite ? '' : '&access_type=offline'}&code_challenge_method=S256&client_id=${IKConstants.CLIENT_ID}&response_type=code&redirect_uri=${redirectTo}`);
    }).catch(() => {
        console.log('[login/utils > getChallengeAndRedirectToLogin] error redirect');
    });
}

/**
 * check if token is expired
 * @returns bool
 */
export function checkIKTokenIsExpired() {
    if (isServerVersionGreaterThanOrEqualTo(getDesktopVersion(), '2.1.0')) {
        return false;
    }

    const tokenExpire = localStorage.getItem('IKTokenExpire');
    if (!tokenExpire) {
        return true;
    }

    const isExpired = parseInt(tokenExpire, 10) <= Date.now() / 1000;

    if (isExpired) {
        console.log('[login/utils > checkIKTokenIsExpired] token is expired');
        localStorage.setItem('tokenExpired', '1');
    }
    return isExpired;
}

/**
 * check if token is expiring in 1 min
 * @returns bool
 */
export function checkIKTokenExpiresSoon(): boolean {
    if (isServerVersionGreaterThanOrEqualTo(getDesktopVersion(), '2.1.0')) {
        return false;
    }

    const tokenExpire = localStorage.getItem('IKTokenExpire');
    if (!tokenExpire) {
        return true;
    }

    const isExpiredInOneMinute = parseInt(tokenExpire, 10) <= ((Date.now() / 1000) + 60);
    return isExpiredInOneMinute;
}

export function isDefaultAuthServer() {
    return window.location.origin === v2DefaultAuthServer;
}

function storeTokenV2(tokenData: {token: string; refreshToken?: string; expiresAt?: number}) {
    localStorage.setItem('IKToken', tokenData.token);
    if (tokenData.refreshToken) {
        localStorage.setItem('IKRefreshToken', tokenData.refreshToken);
    }
    if (tokenData.expiresAt) {
        localStorage.setItem('IKTokenExpire', tokenData.expiresAt);
        localStorage.setItem('tokenExpired', '0');
    }

    Client4.setToken(tokenData.token);
    Client4.setCSRF(tokenData.token);
    Client4.setAuthHeader = true;
}

async function refreshTokenV2(): Promise<{token: string; refreshToken: string; expiresAt: number}> {
    try {
        const newToken = await window.authManager.refreshToken();
        console.log(newToken);
        storeTokenV2(newToken);
        return newToken;
    } catch (error) {
        console.error(error);
        window.postMessage(
            {
                type: 'reset-teams',
                message: {},
            },
            window.origin,
        );
        return Promise.reject(error);
    }
}

// TODO: change types here since it is only used for 2.0 and below tokens which always have expire and refresh.
function isValidTokenV2(token: {token: string; refreshToken?: string; expiresAt?: number}) {
    const isExpiredInOneMinute = token.expiresAt! <= ((Date.now() / 1000) + 60);

    return !isExpiredInOneMinute;
}

export async function refreshIKToken(redirectToTeam = false): Promise<string> {
    const updatedToken: {token: string; refreshToken?: string; expiresAt?: number} = await window.authManager.tokenRequest();

    // If desktop doesn't have a valid token/refresh token
    if (!Object.keys(updatedToken).length) {
        // Clean token storage just in case and reject promise.
        clearLocalStorageToken();
        return Promise.reject(new Error('missing refresh token'));
    } else if (!updatedToken.expiresAt || isValidTokenV2(updatedToken)) {
        // If 2.1 and above and no token expiration, or 2.0 token is valid, use it.
        storeTokenV2(updatedToken);
        return updatedToken.token;
    }

    // Otherwise token is valid but expired, request a refresh.
    try {
        const {token} = await refreshTokenV2();

        // Queue redirect before returning, otherwise it won't trigger.
        if (redirectToTeam) {
            redirectUserToDefaultTeam();
        }

        return token;
    } catch (e) {
        return Promise.reject(e);
    }
}
