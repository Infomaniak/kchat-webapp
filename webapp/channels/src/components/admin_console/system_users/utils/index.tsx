// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type {SortingState} from '@tanstack/react-table';
import React from 'react';
import {FormattedMessage} from 'react-intl';

import {UserReportSortColumns, ReportSortDirection} from '@mattermost/types/reports';
import type {UserReportOptions, UserReport} from '@mattermost/types/reports';
import type {Team} from '@mattermost/types/teams';

import {PAGE_SIZES} from 'components/admin_console/list_table';

import type {AdminConsoleUserManagementTableProperties} from 'types/store/views';

import {ColumnNames, RoleFilters, StatusFilter, TeamFilters} from '../constants';
import type {TableOptions} from '../system_users';
import type {OptionType as TeamFilterOptionType} from '../system_users_filters_popover/system_users_filter_team';

export function convertTableOptionsToUserReportOptions(tableOptions?: TableOptions): UserReportOptions {
    return {
        page_size: tableOptions?.pageSize || PAGE_SIZES[0],
        from_column_value: tableOptions?.fromColumnValue,
        from_id: tableOptions?.fromId,
        direction: tableOptions?.direction,
        ...getSortColumnForOptions(tableOptions?.sortColumn),
        ...getSortDirectionForOptions(tableOptions?.sortIsDescending),
        ...getSearchFilterOption(tableOptions?.searchTerm),
        ...getTeamFilterOption(tableOptions?.filterTeam),
        ...getStatusFilterOption(tableOptions?.filterStatus),
        ...getRoleFilterOption(tableOptions?.filterRole),
        date_range: tableOptions?.dateRange,
    };
}

/**
 * Converts the sorting column name to API compatible sorting column name. Default sorting column name is by username.
 */
export function getSortColumnForOptions(id?: SortingState[0]['id']): Pick<UserReportOptions, 'sort_column'> {
    let sortColumn: UserReportOptions['sort_column'];

    if (id === ColumnNames.email) {
        sortColumn = UserReportSortColumns.email;
    } else if (id === ColumnNames.createAt) {
        sortColumn = UserReportSortColumns.createAt;
    } else {
        // Default sorting to first User details column
        sortColumn = UserReportSortColumns.username;
    }

    return {
        sort_column: sortColumn,
    };
}

/**
 * Converts the sorting direction to API compatible sorting direction. Default sorting direction is ascending.
 */
export function getSortDirectionForOptions(desc?: SortingState[0]['desc']): Pick<UserReportOptions, 'sort_direction'> {
    let sortDirection: UserReportOptions['sort_direction'];

    if (desc) {
        sortDirection = ReportSortDirection.descending;
    } else {
        sortDirection = ReportSortDirection.ascending;
    }

    return {
        sort_direction: sortDirection,
    };
}

/**
 * It returns the value of that column on which sorting is applied.
 */
export function getSortableColumnValueBySortColumn(row: UserReport, sortColumn: AdminConsoleUserManagementTableProperties['sortColumn']): string {
    switch (sortColumn) {
    case ColumnNames.email:
        return row.email;
    case ColumnNames.createAt:
        return String(row.create_at);
    default:
        return row.username;
    }
}

export function getTeamFilterOption(teamId?: string): Partial<Pick<UserReportOptions, 'team_filter' | 'has_no_team'>> {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            team_filter: undefined,
            has_no_team: undefined,
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            team_filter: undefined,
            has_no_team: true,
        };
    }

    return {
        team_filter: teamId,
        has_no_team: undefined,
    };
}

export function getStatusFilterOption(status?: string): Partial<Pick<UserReportOptions, 'hide_active' | 'hide_inactive'>> {
    if (status === StatusFilter.Active) {
        return {
            hide_inactive: true,
        };
    } else if (status === StatusFilter.Deactivated) {
        return {
            hide_active: true,
        };
    }

    return {
        hide_active: undefined,
        hide_inactive: undefined,
    };
}

export function getPaginationInfo(pageIndex: number, pageSize: number, currentLength: number, total?: number) {
    if (!currentLength) {
        return (
            <FormattedMessage
                id='admin.system_users_list.pagination.no_users'
                defaultMessage='0 users'
            />
        );
    }

    const firstPage = (pageIndex * pageSize) + 1;
    const lastPage = (pageIndex * pageSize) + currentLength;
    const totalItems = total || 0;

    return (
        <FormattedMessage
            id='admin.system_users_list.pagination'
            defaultMessage='Showing {firstPage} - {lastPage} of {totalItems} users'
            values={{
                firstPage,
                lastPage,
                totalItems,
            }}
        />
    );
}

export function getDefaultSelectedValueFromList<T extends {value: string}>(value: string, options: T[]) {
    const option = options.find((option) => option.value === value);

    if (option) {
        return option;
    }

    return options[0];
}

export function getDefaultSelectedTeam(teamId: Team['id'] | string, label?: string): TeamFilterOptionType {
    if (!teamId || teamId === TeamFilters.AllTeams) {
        return {
            value: TeamFilters.AllTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.allTeams'
                    defaultMessage='All Teams'
                />
            ),
        };
    } else if (teamId === TeamFilters.NoTeams) {
        return {
            value: TeamFilters.NoTeams,
            label: (
                <FormattedMessage
                    id='admin.system_users.filters.team.noTeams'
                    defaultMessage='No Teams'
                />
            ),
        };
    }

    return {
        value: teamId,
        label: label || '',
    };
}

export function getRoleFilterOption(role?: string): Pick<UserReportOptions, 'role_filter'> {
    if (!role || role === RoleFilters.Any) {
        return {role_filter: undefined};
    }
    return {role_filter: role};
}

export function getSearchFilterOption(search?: string): Pick<UserReportOptions, 'search_term'> {
    if (!search || search.trim().length === 0) {
        return {search_term: undefined};
    }

    return {search_term: search};
}
