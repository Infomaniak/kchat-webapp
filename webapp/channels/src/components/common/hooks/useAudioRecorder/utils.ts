export const getWasmFileURL = () =>
    new URL('wasm-media-encoders/wasm/mp3', import.meta.url);
