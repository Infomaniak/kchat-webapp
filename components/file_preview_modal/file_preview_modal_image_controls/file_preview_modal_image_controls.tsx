// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

/* eslint-disable react-hooks/exhaustive-deps */

import React, {ChangeEvent, memo, useEffect, useState} from 'react';
import {useIntl} from 'react-intl';
import {clamp} from 'lodash';

import {t} from 'utils/i18n';

import {MIN_ZOOM_EXT, ZOOM_EXT} from '../image_preview';
import './file_preview_modal_image_controls.scss';

let zoomInButtonActive = true;
let zoomOutButtonActive = false;

interface Props {
    toolbarZoom: number | string;
    setToolbarZoom: (toolbarZoom: number | string) => void;
}

const FilePreviewModalImageControls: React.FC<Props> = ({toolbarZoom, setToolbarZoom}: Props) => {
    // Initial variables and constants
    // zoom text
    const [zoomText, setZoomText] = useState<string>();
    const [selectedZoomValue, setSelectedZoomValue] = useState<string>();

    const plusSign = <i className='icon-plus'/>;
    const minusSign = <i className='icon-minus'/>;

    // Initialize dropdown values
    const {formatMessage} = useIntl();
    const autoText = formatMessage({id: t('imageToolbarZoomDropdown.automatic'), defaultMessage: 'Automatic'});
    const fitWidthText = formatMessage({id: t('imageToolbarZoomDropdown.fitWidth'), defaultMessage: 'Fit width'});
    const fitHeightText = formatMessage({id: t('imageToolbarZoomDropdown.fitHeight'), defaultMessage: 'Fit height'});

    const zoomLevels = new Map();
    zoomLevels.set('A', {text: autoText, type: 'auto'});
    zoomLevels.set('W', {text: fitWidthText, type: 'auto'});
    zoomLevels.set('H', {text: fitHeightText, type: 'auto'});
    zoomLevels.set('1', {text: '100%', type: 'scale'});
    zoomLevels.set('1.25', {text: '125%', type: 'scale'});
    zoomLevels.set('1.5', {text: '150%', type: 'scale'});
    zoomLevels.set('2', {text: '200%', type: 'scale'});
    zoomLevels.set('3', {text: '300%', type: 'scale'});
    zoomLevels.set('4', {text: '400%', type: 'scale'});
    zoomLevels.set('5', {text: '500%', type: 'scale'});

    const zoomLevelOptions = [];
    for (const [zoomLevelKey, zoomLevel] of zoomLevels) {
        zoomLevelOptions.push(
            <option
                key={zoomLevelKey}
                value={zoomLevelKey}
            >{zoomLevel.text}</option>,
        );
    }

    zoomLevelOptions.push(
        <option
            key={'customZoom'}
            value='customZoom'
            hidden={true}
        >
            {zoomText}
        </option>,
    );

    // Handlers
    // change type to proper type in the future
    const handleZoomDropdown = (event: ChangeEvent<HTMLSelectElement | HTMLInputElement>) => {
        const zoomLevel = event.target;
        if (zoomLevels.get(zoomLevel.value).type === 'scale') {
            setToolbarZoom(parseFloat(zoomLevel.value));
        } else {
            setToolbarZoom(zoomLevel.value);
        }
    };

    const handleZoomIn = () => {
        handleZoomButtons(0.1);
    };
    const handleZoomOut = () => {
        handleZoomButtons(-0.1);
    };
    const handleZoomButtons = (delta: number) => {
        let newToolbarZoom = typeof toolbarZoom === 'string' ? ZOOM_EXT : toolbarZoom;
        newToolbarZoom = Math.round(newToolbarZoom * 10) / 10;
        newToolbarZoom = clamp(newToolbarZoom + delta, MIN_ZOOM_EXT, 5);
        setToolbarZoom(newToolbarZoom === MIN_ZOOM_EXT ? 'A' : newToolbarZoom);

        zoomInButtonActive = newToolbarZoom < 5;
        zoomOutButtonActive = newToolbarZoom > MIN_ZOOM_EXT;
    };

    // Callbacks
    useEffect(() => {
        if (typeof toolbarZoom === 'number') {
            setZoomText(`${Math.round(toolbarZoom * 100)}%`);
            zoomInButtonActive = toolbarZoom < 5;
            zoomOutButtonActive = toolbarZoom > MIN_ZOOM_EXT;
        } else if (toolbarZoom === 'A') {
            zoomInButtonActive = true;
            zoomOutButtonActive = false;
        }

        if (zoomLevels.has(toolbarZoom.toString())) {
            setSelectedZoomValue(toolbarZoom.toString());
        } else {
            setSelectedZoomValue('customZoom');
        }
    }, [toolbarZoom]);

    // Elements
    const zoomDropdown = (
        <select
            onChange={handleZoomDropdown}
            className='image-controls-dropdown'
            defaultValue={'A'}
            value={selectedZoomValue}
        >
            {zoomLevelOptions}
        </select>
    );

    const zoomInButton = (
        <button
            onClick={handleZoomIn}
            className={`image-controls-button ${zoomInButtonActive ? 'active' : 'inactive'}`}
        >
            {plusSign}
        </button>
    );
    const zoomOutButton = (
        <button
            onClick={handleZoomOut}
            className={`image-controls-button ${zoomOutButtonActive ? 'active' : 'inactive'}`}
        >
            {minusSign}
        </button>
    );

    // Render
    return (
        <div className='image-controls'>
            {zoomOutButton}
            {zoomInButton}
            {zoomDropdown}
        </div>
    );
};

export default memo(FilePreviewModalImageControls);
