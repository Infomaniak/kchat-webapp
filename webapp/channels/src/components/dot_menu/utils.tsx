// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import type React from 'react';
import {EventTypes, TELEMETRY_CATEGORIES} from 'utils/constants';

import {trackEvent} from 'actions/telemetry_actions';

export type ChangeEvent = React.KeyboardEvent | React.MouseEvent;

export const trackDotMenuEvent = (e: ChangeEvent, suffix: string): void => {
    if (e.type === EventTypes.CLICK) {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.CLICK + '_' + suffix);
    } else {
        trackEvent(TELEMETRY_CATEGORIES.POST_INFO_MORE, EventTypes.SHORTCUT + '_ ' + suffix);
    }
};
