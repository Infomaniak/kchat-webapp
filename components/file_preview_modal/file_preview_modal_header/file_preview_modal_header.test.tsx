// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import {shallow} from 'enzyme';
import React from 'react';

import {Post} from '@mattermost/types/posts';
import {ZoomValue} from '../file_preview_modal_image_controls/file_preview_modal_image_controls';

import {TestHelper} from '../../../utils/test_helper';

import FilePreviewModalHeader from './file_preview_modal_header';

describe('components/file_preview_modal/file_preview_modal_header/FilePreviewModalHeader', () => {
    const defaultProps = {
        enablePublicLink: false,
        canDownloadFiles: true,
        fileURL: 'http://example.com/img.png',
        filename: 'img.png',
        fileInfo: TestHelper.getFileInfoMock({}),
        fileType: 'image',
        isMobileView: false,
        fileIndex: 1,
        totalFiles: 3,
        toolbarZoom: 'A' as ZoomValue,
        post: {} as Post,
        showPublicLink: false,
        isExternalFile: false,
        setToolbarZoom: jest.fn(),
        onGetPublicLink: jest.fn(),
        handlePrev: jest.fn(),
        handleNext: jest.fn(),
        handleModalClose: jest.fn(),
        content: '',
        canCopyContent: true,
    };

    test('should match snapshot the desktop view', () => {
        const props = {
            ...defaultProps,
        };

        const wrapper = shallow(<FilePreviewModalHeader {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot the mobile view', () => {
        const props = {
            ...defaultProps,
            isMobileView: true,
        };

        const wrapper = shallow(<FilePreviewModalHeader {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with width zoom', () => {
        const props = {
            ...defaultProps,
            toolbarZoom: 'W' as ZoomValue,
        };

        const wrapper = shallow(<FilePreviewModalHeader {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with height zoom', () => {
        const props = {
            ...defaultProps,
            toolbarZoom: 'H' as ZoomValue,
        };

        const wrapper = shallow(<FilePreviewModalHeader {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with scale zoom', () => {
        const props = {
            ...defaultProps,
            toolbarZoom: 1.25 as ZoomValue,
        };

        const wrapper = shallow(<FilePreviewModalHeader {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should match snapshot with arbitrary zoom', () => {
        const props = {
            ...defaultProps,
            toolbarZoom: 1.33 as ZoomValue,
        };

        const wrapper = shallow(<FilePreviewModalHeader {...props}/>);
        expect(wrapper).toMatchSnapshot();
    });
});
