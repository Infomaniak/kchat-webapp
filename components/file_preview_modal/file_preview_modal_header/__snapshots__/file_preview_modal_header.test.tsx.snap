// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/file_preview_modal/file_preview_modal_header/FilePreviewModalHeader should match snapshot the desktop view 1`] = `
<Fragment>
  <div
    className="file-preview-modal-header"
  >
    <Memo(FilePreviewModalInfo)
      filename="img.png"
      post={Object {}}
      showFileName={true}
    />
    <Memo(FilePreviewModalMainNav)
      fileIndex={1}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      totalFiles={3}
    />
    <Memo(FilePreviewModalMainActions)
      canDownloadFiles={true}
      enablePublicLink={false}
      fileInfo={
        Object {
          "clientId": "client_id",
          "create_at": 1,
          "delete_at": 1,
          "extension": "jpg",
          "has_preview_image": true,
          "height": 200,
          "id": "file_info_id",
          "mime_type": "mime_type",
          "name": "name",
          "size": 1,
          "update_at": 1,
          "user_id": "user_id",
          "width": 350,
        }
      }
      fileURL="http://example.com/img.png"
      filename="img.png"
      handleModalClose={[MockFunction]}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      isExternalFile={false}
      isMobileView={false}
      onGetPublicLink={[MockFunction]}
      showOnlyClose={false}
      showPublicLink={false}
      usedInside="Header"
    />
  </div>
  <Memo(FilePreviewModalImageControls)
    setToolbarZoom={[MockFunction]}
    toolbarZoom="A"
  />
</Fragment>
`;

exports[`components/file_preview_modal/file_preview_modal_header/FilePreviewModalHeader should match snapshot the mobile view 1`] = `
<Fragment>
  <div
    className="file-preview-modal-header"
  >
    <Memo(FilePreviewModalMainActions)
      canDownloadFiles={true}
      enablePublicLink={false}
      fileInfo={
        Object {
          "clientId": "client_id",
          "create_at": 1,
          "delete_at": 1,
          "extension": "jpg",
          "has_preview_image": true,
          "height": 200,
          "id": "file_info_id",
          "mime_type": "mime_type",
          "name": "name",
          "size": 1,
          "update_at": 1,
          "user_id": "user_id",
          "width": 350,
        }
      }
      fileURL="http://example.com/img.png"
      filename="img.png"
      handleModalClose={[MockFunction]}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      isExternalFile={false}
      isMobileView={true}
      onGetPublicLink={[MockFunction]}
      showOnlyClose={true}
      showPublicLink={false}
      usedInside="Header"
    />
    <Memo(FilePreviewModalMainNav)
      fileIndex={1}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      totalFiles={3}
    />
  </div>
  <Memo(FilePreviewModalImageControls)
    setToolbarZoom={[MockFunction]}
    toolbarZoom="A"
  />
</Fragment>
`;

exports[`components/file_preview_modal/file_preview_modal_header/FilePreviewModalHeader should match snapshot with arbitrary zoom 1`] = `
<Fragment>
  <div
    className="file-preview-modal-header"
  >
    <Memo(FilePreviewModalInfo)
      filename="img.png"
      post={Object {}}
      showFileName={true}
    />
    <Memo(FilePreviewModalMainNav)
      fileIndex={1}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      totalFiles={3}
    />
    <Memo(FilePreviewModalMainActions)
      canDownloadFiles={true}
      enablePublicLink={false}
      fileInfo={
        Object {
          "clientId": "client_id",
          "create_at": 1,
          "delete_at": 1,
          "extension": "jpg",
          "has_preview_image": true,
          "height": 200,
          "id": "file_info_id",
          "mime_type": "mime_type",
          "name": "name",
          "size": 1,
          "update_at": 1,
          "user_id": "user_id",
          "width": 350,
        }
      }
      fileURL="http://example.com/img.png"
      filename="img.png"
      handleModalClose={[MockFunction]}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      isExternalFile={false}
      isMobileView={false}
      onGetPublicLink={[MockFunction]}
      showOnlyClose={false}
      showPublicLink={false}
      usedInside="Header"
    />
  </div>
  <Memo(FilePreviewModalImageControls)
    setToolbarZoom={[MockFunction]}
    toolbarZoom={1.33}
  />
</Fragment>
`;

exports[`components/file_preview_modal/file_preview_modal_header/FilePreviewModalHeader should match snapshot with height zoom 1`] = `
<Fragment>
  <div
    className="file-preview-modal-header"
  >
    <Memo(FilePreviewModalInfo)
      filename="img.png"
      post={Object {}}
      showFileName={true}
    />
    <Memo(FilePreviewModalMainNav)
      fileIndex={1}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      totalFiles={3}
    />
    <Memo(FilePreviewModalMainActions)
      canDownloadFiles={true}
      enablePublicLink={false}
      fileInfo={
        Object {
          "clientId": "client_id",
          "create_at": 1,
          "delete_at": 1,
          "extension": "jpg",
          "has_preview_image": true,
          "height": 200,
          "id": "file_info_id",
          "mime_type": "mime_type",
          "name": "name",
          "size": 1,
          "update_at": 1,
          "user_id": "user_id",
          "width": 350,
        }
      }
      fileURL="http://example.com/img.png"
      filename="img.png"
      handleModalClose={[MockFunction]}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      isExternalFile={false}
      isMobileView={false}
      onGetPublicLink={[MockFunction]}
      showOnlyClose={false}
      showPublicLink={false}
      usedInside="Header"
    />
  </div>
  <Memo(FilePreviewModalImageControls)
    setToolbarZoom={[MockFunction]}
    toolbarZoom="H"
  />
</Fragment>
`;

exports[`components/file_preview_modal/file_preview_modal_header/FilePreviewModalHeader should match snapshot with scale zoom 1`] = `
<Fragment>
  <div
    className="file-preview-modal-header"
  >
    <Memo(FilePreviewModalInfo)
      filename="img.png"
      post={Object {}}
      showFileName={true}
    />
    <Memo(FilePreviewModalMainNav)
      fileIndex={1}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      totalFiles={3}
    />
    <Memo(FilePreviewModalMainActions)
      canDownloadFiles={true}
      enablePublicLink={false}
      fileInfo={
        Object {
          "clientId": "client_id",
          "create_at": 1,
          "delete_at": 1,
          "extension": "jpg",
          "has_preview_image": true,
          "height": 200,
          "id": "file_info_id",
          "mime_type": "mime_type",
          "name": "name",
          "size": 1,
          "update_at": 1,
          "user_id": "user_id",
          "width": 350,
        }
      }
      fileURL="http://example.com/img.png"
      filename="img.png"
      handleModalClose={[MockFunction]}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      isExternalFile={false}
      isMobileView={false}
      onGetPublicLink={[MockFunction]}
      showOnlyClose={false}
      showPublicLink={false}
      usedInside="Header"
    />
  </div>
  <Memo(FilePreviewModalImageControls)
    setToolbarZoom={[MockFunction]}
    toolbarZoom={1.25}
  />
</Fragment>
`;

exports[`components/file_preview_modal/file_preview_modal_header/FilePreviewModalHeader should match snapshot with width zoom 1`] = `
<Fragment>
  <div
    className="file-preview-modal-header"
  >
    <Memo(FilePreviewModalInfo)
      filename="img.png"
      post={Object {}}
      showFileName={true}
    />
    <Memo(FilePreviewModalMainNav)
      fileIndex={1}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      totalFiles={3}
    />
    <Memo(FilePreviewModalMainActions)
      canDownloadFiles={true}
      enablePublicLink={false}
      fileInfo={
        Object {
          "clientId": "client_id",
          "create_at": 1,
          "delete_at": 1,
          "extension": "jpg",
          "has_preview_image": true,
          "height": 200,
          "id": "file_info_id",
          "mime_type": "mime_type",
          "name": "name",
          "size": 1,
          "update_at": 1,
          "user_id": "user_id",
          "width": 350,
        }
      }
      fileURL="http://example.com/img.png"
      filename="img.png"
      handleModalClose={[MockFunction]}
      handleNext={[MockFunction]}
      handlePrev={[MockFunction]}
      isExternalFile={false}
      isMobileView={false}
      onGetPublicLink={[MockFunction]}
      showOnlyClose={false}
      showPublicLink={false}
      usedInside="Header"
    />
  </div>
  <Memo(FilePreviewModalImageControls)
    setToolbarZoom={[MockFunction]}
    toolbarZoom="W"
  />
</Fragment>
`;
